src/replica/replica_throttle.cpp

```diff
@@ -14,40 +14,49 @@
 namespace dsn {
 namespace replication {
 
-bool replica::throttle_request(throttling_controller &controller,
-                               message_ex *request,
-                               int32_t request_units)
+#define THROTTLE_REQUEST(op_type, throttling_type, request, request_units)                         \
+    do {                                                                                           \
+        int64_t delay_ms = 0;                                                                      \
+        auto type = _##op_type##_##throttling_type##_throttling_controller.control(                \
+            request->header->client.timeout_ms, request_units, delay_ms);                          \
+        if (type != throttling_controller::PASS) {                                                 \
+            if (type == throttling_controller::DELAY) {                                            \
+                tasking::enqueue(                                                                  \
+                    LPC_THROTTLING_DELAY,                                                          \
```

Comment:
```suggestion
                    LPC_THROTTLING_DELAY,                                                          \
```

`LPC_WRITE_THROTTLING_DELAY` was previously in the replication thread pool. If read is also throttled, it should be retried in `THREAD_POOL_LOCAL_APP` thread pool.

include/dsn/dist/replication/replication.codes.h

```diff
// THREAD_POOL_LOCAL_APP
#define CURRENT_THREAD_POOL THREAD_POOL_LOCAL_APP
MAKE_EVENT_CODE(LPC_WRITE, TASK_PRIORITY_COMMON)
+MAKE_EVENT_CODE(LPC_READ_THROTTLING, TASK_PRIORITY_COMMON)
#undef CURRENT_THREAD_POOL
```